docs/src/content/docs/book/functions.mdx

```diff
@@ -64,33 +240,96 @@ fun customPow(a: Int, c: Int): Int {
 }
 ```
 
-## Virtual and abstract functions
+#### Member functions {#fun-member}
 
-You can allow a contract inheriting a [trait](/book/types#traits) to modify an internal function marked with the `virtual{:tact}` keyword by using `override{:tact}`. A function can also be marked as `abstract{:tact}`, in which case the inheriting contract must define its implementation:
+<Badge text="Contract scope" variant="note" size="medium"/><p/>
 
-```tact
-trait FilterTrait with Ownable {
-    // Virtual functions can be overridden by users of this trait
-    virtual fun filterMessage(): Bool {
-        return sender() != self.owner;
-    }
+These functions behave similarly to private methods in popular object-oriented languages — they're internal to contracts and can be called by prefixing them with a special [identifier `self{:tact}`](/book/contracts#self). That's why member functions are sometimes referred to as "contract methods".
 
-    abstract fun specialFilter(): Bool;
-}
+Member functions can access the contract's [persistent state variables](/book/contracts#variables) and [constants](/book/contracts#constants).
 
-contract Filter with FilterTrait {
-    // Overriding the default behavior of the FilterTrait
-    override fun filterMessage(): Bool {
-        return true;
-    }
+They can only be called from [receivers](#receive), [getters](#get), and other member functions, but not from other contracts.
 
-    override fun specialFilter(): Bool {
-        return true;
+### `init` — constructor function {#init}
```

Comment:
To better explain `init`, add an example with contract parameters (including an `initialized: Bool` flag) and show how `init` can be simulated. This can be done using the Counter contract and its two equivalent implementations using contract fields and contract parameters.